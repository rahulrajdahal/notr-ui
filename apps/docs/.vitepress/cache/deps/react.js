import {
  require_react
} from "./chunk-UUF3TEGA.js";
import "./chunk-5WRI5ZAA.js";
export default require_react();
//# sourceMappingURL=react.js.map
