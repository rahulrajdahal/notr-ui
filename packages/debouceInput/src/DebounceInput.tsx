import { css } from "@emotion/react";
import React from "react";

const baseStyles = css`
  outline: none;
  border: none;
`;

export interface DebounceInputProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "onChange"> {
  /** Value of the input. */
  value: string | number;
  /** OnChange handler. Update value when changed.*/
  onChange: (value: string | number) => void;
  /** time for input delay in milliseconds.*/
  debounce?: number;
}

const DebounceInput = React.forwardRef<HTMLInputElement, DebounceInputProps>(
  ({ debounce, value: initialValue, onChange, ...props }, ref) => {
    const [value, setValue] = React.useState(initialValue);

    React.useEffect(() => setValue(initialValue), [initialValue]);

    React.useEffect(() => {
      const timeout = setTimeout(() => {
        onChange(value);
      }, debounce);

      return () => clearTimeout(timeout);
    }, [debounce, onChange, value]);

    return (
      <input
        ref={ref}
        value={value}
        onChange={(e) => setValue(e.target.value)}
        css={baseStyles}
        {...props}
      />
    );
  }
);

DebounceInput.displayName = "DebounceInput";

export default DebounceInput;
