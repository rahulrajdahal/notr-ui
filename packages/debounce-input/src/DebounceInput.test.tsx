import { createSerializer } from "@emotion/jest";
import { fireEvent, render, screen } from "@testing-library/react";
import { vi } from "vitest";
import DebounceInput from "./DebounceInput";

describe("Debounce Input Component", () => {
  it("renders with default props and styles", () => {
    let value = "";

    render(
      <DebounceInput
        data-testid="notr-debounce"
        value={value}
        onChange={(value) => (value = value)}
      />
    );
    const input = screen.getByTestId("notr-debounce");
    expect(input).toBeInTheDocument();
    expect(input).toHaveStyle(`border: none`);
    expect(input).toHaveStyle(`outline: none`);

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();

    fireEvent.change(input, { target: { value: "Debounce Input" } });
    expect(input.value).toBe("Debounce Input");

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();
  });

  it("renders with delay props and styles", () => {
    let value = "";
    vi.useFakeTimers();
    const mockFn = vi.fn();

    render(
      <DebounceInput
        data-testid="notr-debounce"
        value={""}
        onChange={mockFn}
        debounce={1000}
      />
    );
    const input = screen.getByTestId("notr-debounce");
    expect(input).toBeInTheDocument();
    expect(input).toHaveStyle(`border: none`);
    expect(input).toHaveStyle(`outline: none`);
    fireEvent.change(input, { target: { value: "Debounce" } });
    fireEvent.change(input, { target: { value: "Debounce value" } });
    expect(mockFn).not.toHaveBeenCalled();

    vi.advanceTimersByTime(1000);
    expect(mockFn).toHaveBeenCalledTimes(1);
    expect(mockFn).toHaveBeenCalledWith("Debounce value");
    expect(input.value).toBe("Debounce value");
    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();

    fireEvent.change(input, { target: { value: "Call again?" } });
    expect(mockFn).toHaveBeenCalledTimes(1);

    vi.advanceTimersByTime(1000);
    expect(mockFn).toHaveBeenCalledTimes(2);
    expect(mockFn).toHaveBeenCalledWith("Call again?");
  });
});
