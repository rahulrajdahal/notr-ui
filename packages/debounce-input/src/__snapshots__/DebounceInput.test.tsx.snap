// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Debounce Input Component > renders with default props and styles 1`] = `
.emotion-0 {
  outline: none;
  border: none;
}

<input
  class="emotion-0"
  data-testid="notr-debounce"
  value=""
/>
`;

exports[`Debounce Input Component > renders with default props and styles 2`] = `
.emotion-0 {
  outline: none;
  border: none;
}

.emotion-0 {
  outline: none;
  border: none;
}

<input
  class="emotion-0"
  data-testid="notr-debounce"
  value="Debounce Input"
/>
`;

exports[`Debounce Input Component > renders with delay props and styles 1`] = `
.emotion-0 {
  outline: none;
  border: none;
}

.emotion-0 {
  outline: none;
  border: none;
}

.emotion-0 {
  outline: none;
  border: none;
}

<input
  class="emotion-0"
  data-testid="notr-debounce"
  value="Debounce value"
/>
`;
